src/ddmd/traits.d

```diff
@@ -1353,6 +1353,22 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
         if (auto fa2 = s2.isFuncAliasDeclaration())
             s2 = fa2.toAliasFunc();
 
+        // https://issues.dlang.org/show_bug.cgi?id=11259
+        // compare import symbol to a package symbol
+        for(int i = 0; i < 2; i++)
+        {
+            // if i == 0, compare s1 (import) to s2 (package)
+            // else compare s1(package) to s2 (import)
+            if (auto imp = i == 0 ? s1.isImport() : s2.isImport())
+            {
+                if (auto pkg = i == 0 ? s2.isPackage() : s1.isPackage())
+                {
+                    if (imp.pkg == pkg)
+                        return True();
```

Comment:
Right you are. My mistake. How about this:
```
static bool cmp(Dsymbol s1, Dsymbol s2)
{
    auto imp = s1.isImport();
    return imp && imp.pkg && imp.pkg == s2.isPackage();
}

if (cmp(s1,s2) || cmp(s2,s1))
    return True();
```
I've used the loop method in C/C++ when nested functions weren't available. I think nested functions are nicer.